lib/src/current_weather/data/repository/current_weather_repository.dart

```diff
@@ -1,16 +1,28 @@
+import 'package:connectivity_plus/connectivity_plus.dart';
+
+import '../../../../errors/generic_server_exception.dart';
 import '../../../locations/domain/entities/current_location.dart';
 import '../../domain/entities/current_weather.dart';
 import '../../domain/repository/current_weather_repository_interface.dart';
 import '../adapters/current_weather_from_dto.dart';
 import '../models/current_weather_dto.dart';
 import '../sources/current_weather_api.dart';
+import '../sources/current_weather_local.dart';
 
 class CurrentWeatherRepository implements CurrentWeatherRepositoryInterface {
-  CurrentWeatherRepository(this.api);
+  CurrentWeatherRepository(this.api, this.local, this.connectivity);
   final CurrentWeatherApi api;
+  final CurrentWeatherLocal local;
+  final Connectivity connectivity;
 
   @override
   Future<CurrentWeather> getCurrentWeather(CurrentLocation location) async {
+    final connection = await connectivity.checkConnectivity();
+    final hasInternet = connection != ConnectivityResult.none;
+    if (!hasInternet) {
+      throw GenericServerException();
```

Comment:
I'd ask you to kindly define and implement a `NoConnectionAvailable` exception, instead.